File: server/models/Teacher.js
const mongoose = require('mongoose');

const TeacherSchema = new mongoose.Schema({
    fullName: {
        type: String,
        required: [true, 'ФИО преподавателя обязательно'],
        trim: true
    },
    phone: {
        type: String,
        trim: true
    },
    subjects: [{
        type: mongoose.Schema.Types.ObjectId,
        ref: 'Subject'
    }],
    isActive: {
        type: Boolean,
        default: true
    }
}, {
    timestamps: true
});

module.exports = mongoose.model('Teacher', TeacherSchema);

File: lib/features/admin/domain/entities/teacher.dart
import 'package:freezed_annotation/freezed_annotation.dart';

part 'teacher.freezed.dart';
part 'teacher.g.dart';

@freezed
class Teacher with _$Teacher {
  const factory Teacher({
    String? id,
    required String fullName,
    String? phone,
    @Default([]) List<String> subjects,
    @Default(true) bool isActive,
  }) = _Teacher;

  factory Teacher.fromJson(Map<String, dynamic> json) => _$TeacherFromJson(json);
}
